src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1873,4 +1881,104 @@ private boolean expectedFunction(@Nullable PsiElement resolve) {
       return true;
     }
   }
+
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> NAMES;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      NAMES = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom atom = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      ErlangMaxExpression expr = ObjectUtils.tryCast(atom != null ? atom.getParent() : null, ErlangMaxExpression.class);
+      ErlangArgumentList args = ObjectUtils.tryCast(expr != null ? expr.getParent() : null, ErlangArgumentList.class);
+      if (expr == null || args == null) return false;
+
+      int position = args.getExpressionList().indexOf(expr);
+      if (position == -1) return false;
+
+      ErlangFunctionCallExpression call = ObjectUtils.tryCast(args.getParent(), ErlangFunctionCallExpression.class);
+      ErlangFunction function = resolveToFunction(call);
+      if (function == null) return false;
+
+      for (ErlangFunctionClause clause : function.getFunctionClauseList()) {
+        if (elementSuits(clause, position)) return true;
```

Comment:
It would be a lot easier to read if you extracted a method which would get parameter name and check it's in your names list right here.